### DISCUSSION TITLE
Fix for Amazon FireOS devices

### DISCUSSION THREAD

**Discussion by emtee40** [d4rken-org/sdmaid-se#903] [Show and tell]
It's slow when cleaning cache, but it works. 

**Accessibility Service**
First get this app mentioned [here.](https://xdaforums.com/t/app-firetv-accessibility-permissions-manager-for-firetv-devices.4144057/) It runs very annoyingly and very similar to Shizuku (ie: having to reopen app to start service for requesting app..). This will grant Accessibility Service.

For the next steps Remote ADB Shell is needed. Find it [here.](https://github.com/cgutman/RemoteAdbShell/releases)

**Grant Usage Permission**
Open Remote ADB and connect.
Run this command:
`pm grant eu.darken.sdmse android.permission.PACKAGE_USAGE_STATS`

**Grant Drawing over other apps**
Open Remote ADB and connect.
Run this command:
`pm grant eu.darken.sdmse android.permission.SYSTEM_ALERT_WINDOW`

**Extra suggestion**
Use Remote ADB again and run this:
`settings put global hidden_api_policy 1`

Since FireOS is based off Android 9 maybe, I ran both commands:
`settings put global hidden_api_policy_p_apps 1`

**Conclusion**
It works. Change out package name for OG _SD Maid_ and it will run as well. Both verified. Just remember that first line of this message: Very, VERY, slow when cleaning app Caches.

**Comment by d4rken** [comment]
What makes it slow? Does SD Maid need multiple retries? Or does the FireOS UI react slow?

Can you make a video that demos it?
